be2-scala/src/test/scala/ch/epfl/pop/model/network/JsonRpcResponseSuite.scala

```diff
@@ -0,0 +1,31 @@
+package ch.epfl.pop.model.network
+
+import org.scalatest.{FunSuite, Matchers}
+
+class JsonRpcResponseSuite extends FunSuite with Matchers {
+    test("Constructor/apply works as intended"){
+        val rpc: String = "rpc"
+        val res: Option[ResultObject] = Some(new ResultObject(2))
+        val err: Option[ErrorObject] = Some(new ErrorObject(1, "Error"))
+        val id: Option[Int] = Some(0)
+
+        val response: JsonRpcResponse = JsonRpcResponse(rpc, res, err, id)
+        response.jsonrpc should equal(rpc)
+        response.result should equal(res)
+        response.error should equal(err)
+        response.id should equal(id)
```

Comment:
Interestingly enough, having an error field and a result field should not be legal. It is either one or the other. A JsonRpcResponse like this would not be accepted by front-ends. Could you add a simple check that prevents having both fields in the apply?

Reply:
of course